### DISCUSSION TITLE
Is it possible to have multiple authentication with different table/collection

### DISCUSSION THREAD

**Discussion by Rem0ld** [feathersjs/feathers#2624] [Q&A] [answered]
Hello,

Let me explain, when you create a feathers app, we use the User collection (or table for SQL) to do the authentication.
So far so good.
Today we have users that can connect through this authentication but we would like to authorize another collection, let's call it "Talent" to have a different authentication.
So I want my Users to have certain access (already configured and working) with authentication 1
And I want Talents to have authentication for different services with authentication 2.

Authentication 2 seems to be ready, I can create a Talent, then I can make a call on "/talent-authentication" and I'm getting a JWT + talent information.

But when I'm trying to protect services with authentication 2, I'm getting this error

```
NotAuthenticated: Invalid authentication information (no `strategy` set)
```

I configured default.json
```
...
"authentication": {
    "entity": "user",
    "service": "users",
    "secret": "...",
    "authStrategies": [
      "jwt",
      "local"
    ],
    "passwordPolicy": {
     ...
    },
    "maximumSessionLength": "1d",
    "jwtOptions": {
      "header": {
        "typ": "access"
      },
      "audience": "https://yourdomain.com",
      "issuer": "feathers",
      "algorithm": "HS256",
      "expiresIn": "8h"
    },
    "local": {
      "usernameField": "login",
      "passwordField": "password"
    }
  },
"talentAuthentication": {
    "entity": "loginTalent",
    "service": "login-talent",
    "secret": "...",
    "authStrategies": [
      "jwt",
      "talent",
      "local"
    ],
    "passwordPolicy": {...},
    "local": {
      "usernameField": "email",
      "passwordField": "password"
    },
    "jwtOptions": {
      "header": {
        "typ": "access"
      },
      ...
    }
  },
```

and in the hook's service I wanna protect

```
const authenticateTalent = () => (context) =>  {
  if (params.provider && !params.authentication) {
      context.params = {
        ...params,
        authentication: {
          strategy: 'talent',
        },
      }
  }

  return context
}
 before: {
    all: [authenticateTalent(), authenticate('jwt', 'talent')],
    ...
  },

```

If someone can tell me if what I'm trying to do is possible and if so how to do it.
Thank you

**Comment by daffl** [comment, marked as answer]
The format for the `authenticate` hook to use a different authentication service (https://docs.feathersjs.com/api/authentication/hook.html#authenticate-options) is:

```js
authenticate({
  strategies: [ 'jwt' ],
  service: 'talent-authentication'
})
```

**Reply by Rem0ld** [reply]
I found this repo which have a very good explanation on how to do things
[https://github.com/ShinChven/multiple-authentication-services-on-feathers-app](https://github.com/ShinChven/multiple-authentication-services-on-feathers-app)
However I followed it at the letter and I'm still getting this `no strategy set error`
When I log in I'm getting a  200 response with the token and I can still access the protected route without token.
And the server logs the error which I don't quite understand.

Is it supposed to work with this configuration, that's all I'm asking ^^
Because we have some additional configuration which could lead to weird behaviour and I don't want to waste your time.

**Reply by Rem0ld** [reply]
I finally discover what was my problem, my editor was importing the first line and not the second one
```
const { authenticate } = require('@feathersjs/express')
const { authenticate } = require('@feathersjs/authentication').hooks
```
